node/src/BaseClient.ts

```diff
@@ -4841,9 +4849,15 @@ export class BaseClient {
      * ```
      */
     public async xread(
-        keys_and_ids: Record<string, string>,
+        keys_and_ids: Record<string, GlideString> | GlideRecord<GlideString>,
         options?: StreamReadOptions,
     ): Promise<Record<string, Record<string, [string, string][]>>> {
+        if (!Array.isArray(keys_and_ids)) {
```

Comment:
can you explain this part?

Reply:
Basically a type conversion. Have moved to an auxiliary function to avoid code duplication, as Yury suggested.